src/pydicom/dataelem.py

```diff
@@ -449,6 +482,48 @@ def value(self, val: Any) -> None:
 
         self._value = self._convert_value(val)
 
+    
+    def value_generator(self, *, buffer_info: BufferInfo = None, chunk_size = 1024) -> Iterator[bytes]:
+        """Consume data from a buffered value. The value must be a buffer, the buffer must be
```

Comment:
I think probably at least 8K chunk default is better.  We had this come up with pydicom's undefined length reads before (#436).

But I'm also now wondering whether this generator is needed.  Any code using buffered reading can just directly use `value` after checking it is a buffered reader.  I need to ponder a little more ...